src/main/java/org/usfirst/frc1745/deepspace2019/Robot.java

```diff
@@ -103,17 +111,48 @@ public void teleopInit() {
         // teleop starts running. If you want the autonomous to
         // continue until interrupted by another command, remove
         // this line or comment it out.
-        if (autonomousCommand != null) autonomousCommand.cancel();
-        
+        if (autonomousCommand != null)
+            autonomousCommand.cancel();
+
     }
 
     /**
      * This function is called periodically during operator control
      */
     @Override
     public void teleopPeriodic() {
-        Scheduler.getInstance().run(); 
+        Scheduler.getInstance().run();
         drive.setLeftSpeed(-controls.getLeftY(DEADZONE));
         drive.setRightSpeed(controls.getRightY(DEADZONE));
+
+        //Adds the right and left delta values to SmartDashboard
+        NetworkTable limelightNetworkTable = limelight.getLimelightNetworkTable();
+        double[] calculatedDeltas = limelight.calcSpeed(limelightNetworkTable);
+        
+        //Send all data to the dashboard
+        SmartDashboard.putNumber("Left Delta: ", calculatedDeltas[0]);
+        SmartDashboard.putNumber("Right Delta ", calculatedDeltas[1]);
+        SmartDashboard.putNumber("Steering Adjust:", calculatedDeltas[2]);
+        SmartDashboard.putNumber("Minimum Aim Constant: ", limelight.getMinAimCommand());
+        SmartDashboard.putNumber("Distance Constant: ", limelight.getKpDistance());
+        SmartDashboard.putNumber("Aim Constant: ", limelight.getKpAim());
+        SmartDashboard.putNumber("Tx Value:", limelightNetworkTable.getEntry("tx").getDouble(0.0));
+        SmartDashboard.putNumber("Ty Value:", limelightNetworkTable.getEntry("ty").getDouble(0.0));
+        SmartDashboard.putNumber("Deadband", limelight.getDeadband());
+       
+        //Set constants
+        Preferences preferences = Preferences.getInstance();
+        limelight.setKpAim(preferences.getDouble("KpAimConstant", limelight.getKpAim()));
+        limelight.setKpDistance(preferences.getDouble("KpDistanceConstant", limelight.getKpDistance()));
+        limelight.setMinAimCommand(preferences.getDouble("MinAimCommand", limelight.getMinAimCommand()));
+        limelight.setDeadband(preferences.getDouble("Deadband", limelight.getDeadband()));
```

Comment:
Looking at `preferences.getDouble(string, double)` it says:
`Returns the double at the given key. If this table does not have a value for that position,
   * then the given backup value will be returned.`

The double passed in is the "backup value", we should set the backup value to 0.0 instead of the limelight values because it will be easier to debug a 0.0